torch/csrc/jit/argument_spec.cpp

```diff
@@ -158,7 +158,7 @@ ArgumentSpec ArgumentSpecCreator::create(bool with_grad, const Stack& input)
         AT_ASSERT(iv->isTuple());
         // see [argspec refcounting]
         auto p = *reinterpret_cast<const at::ivalue::Tuple* const*>(iv);
-        auto tup_ptr = &p->elements()[0];
+        auto tup_ptr = c10::impl::ptr_to_first_element(p->elements());
```

Comment:
why is it a separate function?

Reply:
because `operator[]` doesn't return something that you can get a pointer from.